drivers/bridge/port_mapping.go

```diff
@@ -11,71 +11,113 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
-var (
-	defaultBindingIP   = net.IPv4(0, 0, 0, 0)
-	defaultBindingIPV6 = net.ParseIP("::")
-)
-
 func (n *bridgeNetwork) allocatePorts(ep *bridgeEndpoint, reqDefBindIP net.IP, ulPxyEnabled bool) ([]types.PortBinding, error) {
 	if ep.extConnConfig == nil || ep.extConnConfig.PortBindings == nil {
 		return nil, nil
 	}
 
-	defHostIP := defaultBindingIP
+	defHostIP := net.IPv4zero // 0.0.0.0
 	if reqDefBindIP != nil {
 		defHostIP = reqDefBindIP
 	}
 
-	// IPv4 port binding including user land proxy
-	pb, err := n.allocatePortsInternal(ep.extConnConfig.PortBindings, ep.addr.IP, defHostIP, ulPxyEnabled)
-	if err != nil {
-		return nil, err
+	var containerIPv6 net.IP = nil
+	if ep.addrv6 != nil {
+		containerIPv6 = ep.addrv6.IP
 	}
 
-	// IPv6 port binding excluding user land proxy
-	if n.driver.config.EnableIP6Tables && ep.addrv6 != nil {
```

Comment:
The `EnableIP6Tables` check was removed completely.

Is there anything that prevents a creation of ip6tables rules in `AppendForwardingTableEntry` of the portmapper?

Reply:
If https://github.com/moby/libnetwork/blob/5c6a95bfb20c61571a00f913c6b91959ede84e8d/drivers/bridge/setup_ip_tables.go#L184
 is skipped, `pm.chain` will be `nil` and will skip the iptables programming
https://github.com/moby/libnetwork/blob/5c6a95bfb20c61571a00f913c6b91959ede84e8d/portmapper/mapper_linux.go#L42

Reply:
oh I see. That is for sure far cleaner than before.